py/server/deephaven/updateby.py

```diff
@@ -1360,4 +1358,90 @@ def rolling_wavg_time(ts_col: str, wcol: str, cols: Union[str, List[str]], rev_t
         fwd_time = _JDateTimeUtils.parseDurationNanos(fwd_time) if isinstance(fwd_time, str) else fwd_time
         return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingWAvg(ts_col, rev_time, fwd_time, wcol, *cols))
     except Exception as e:
-        raise DHError(e, "failed to create a rolling weighted average (time) UpdateByOperation.") from e
\ No newline at end of file
+        raise DHError(e, "failed to create a rolling weighted average (time) UpdateByOperation.") from e
+
+
+def rolling_formula_tick(formula: str, formula_param: str, cols: Union[str, List[str]], rev_ticks: int, fwd_ticks: int = 0) -> UpdateByOperation:
+    """Creates a rolling formula UpdateByOperation for the supplied column names, using ticks as the windowing unit. Ticks
+    are row counts, and you may specify the reverse and forward window in number of rows to include. The current row
+    is considered to belong to the reverse window but not the forward window. Also, negative values are allowed and
+    can be used to generate completely forward or completely reverse windows.
+
+    Here are some examples of window values:
+        |  `rev_ticks = 1, fwd_ticks = 0` - contains only the current row
+        |  `rev_ticks = 10, fwd_ticks = 0` - contains 9 previous rows and the current row
+        |  `rev_ticks = 0, fwd_ticks = 10` - contains the following 10 rows, excludes the current row
+        |  `rev_ticks = 10, fwd_ticks = 10` - contains the previous 9 rows, the current row and the 10 rows following
+        |  `rev_ticks = 10, fwd_ticks = -5` - contains 5 rows, beginning at 9 rows before, ending at 5 rows before  the
+            current row (inclusive)
+        |  `rev_ticks = 11, fwd_ticks = -1` - contains 10 rows, beginning at 10 rows before, ending at 1 row before the
+            current row (inclusive)
+        |  `rev_ticks = -5, fwd_ticks = 10` - contains 5 rows, beginning 5 rows following, ending at 10 rows  following the
+            current row (inclusive)
+
+    Args:
+        formula (str): the user defined formula to apply to each window of data
+        formula_param (str): the parameter name within the formula
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the rolling formula operation on all columns.
```

Comment:
This signature looks very locked into a single column in / out implementation.  Is the plan to have a multiple in implementation have a different function?  If not, this function signature should be evaluated.

Reply:
This matches the `AggBy` `Formula` syntax which was the primary goal of this PR.  The multiple column syntax is TBD for both `Agg` and `UpdateBy` formula operators.

Reply:
Sounds good.